### PR TITLE
WIP: better docstrings and signatures for decorators

### PR DESCRIPTION
**Pull request by aflaxman**
For better UX in Jupyter notebooks, I've added some code to change the signature of `uses_columns`.  It does not look quite right, and is a bit nasty, so maybe someone will help me do it better.

### CODE REVIEW

File: vivarium/framework/population.py [hunk 1]
```diff
@@ -10,18 +10,31 @@
...
 query   : str
-          A filter in pandas query syntax which should be applied to the population before it made accessible
-          to this function. This effects both the ``population`` and the ``index`` attributes of PopulationEvents
+          A filter in pandas query syntax which should be applied to
+          the population before it made accessible to this
+          function. This effects both the ``population`` and the
+          ``index`` attributes of PopulationEvents
 """
+from inspect import signature, Parameter, Signature
+uses_columns.__signature__ = \
```
Comment (alec-deason) [inline vivarium/framework/population.py]: If you want a more tool friendly signature I wouldn't do this. Rather I would wrap the generated function like this:
`
_uses_columns = resource_injector('population_system_population_view')
def uses_columns(columns, query):
    '''DOCSTRING'''
   return _uses_columns(columns, query)
`

I haven't actually tried that but it should work and give you the behavior you want.

**Comment by aflaxman** [comment]
That is way simpler, but doesn't quite work.  Cool that travis can show you how: 
```
___________________________ test_interpolated_tables ___________________________
    def test_interpolated_tables():
        years = build_table(lambda age, sex, year: year)
        ages = build_table(lambda age, sex, year: age)
        one_d_age = ages.copy()
        del one_d_age['year']
        one_d_age = one_d_age.drop_duplicates()
    
>       simulation = setup_simulation([generate_test_population], 10000)
tests/framework/test_lookup.py:26: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
vivarium/test_util.py:15: in setup_simulation
    simulation.setup()
vivarium/framework/engine.py:89: in setup
    self.population.setup_components(components)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
self = PopulationManager(_population= Empty DataFrame
Columns: []
Index: [], growing= False, observers= defaultdict(<class 'set'>, {}))
components = [ValuesManager(_pipelines= []), EventManager(event_types: defaultdict(<function EventManager.__init__.<locals>.<lambda...InterpolatedDataManager(clock= <function Builder.__init__.<locals>.<lambda>.<locals>.<lambda> at 0x7f573215aa60>), ...]
    def setup_components(self, components):
>       uses_columns.set_injector(self._population_view_injector)
E       AttributeError: 'function' object has no attribute 'set_injector'
vivarium/framework/population.py:295: AttributeError
```

https://travis-ci.org/ihmeuw/vivarium/jobs/262439596

**Comment by aflaxman** [comment]
This seems safe to merge.  I'm going to do it.

**Comment by aflaxman** [comment]
Oh, I can't!  @alec-deason can you merge this today, or give me feedback on what to change?

**Pull request merged**